### PR TITLE
fix(release): close production rollout blockers

### PR DESCRIPTION
**Pull request by saldanaj97**
## Summary

- prevent past-due Clerk Billing items from promoting paid entitlements
- add free and paid reconciliation regressions
- bind the durable Production migration workflow to the existing `Production – atlaris` environment
- align the architecture contract and deployment documentation

## Verification

- 49 focused unit tests passed
- `pnpm check:type` passed
- focused Oxlint and Oxfmt checks passed
- `bash -n scripts/db/run-phased-migrations.sh` passed
- `git diff --check` passed

## Release impact

This must land on `develop` before PR #428. After it lands, the final release SHA will be used by a one-time, EXPAND-only Production migration bootstrap before the application merge.

---

> [!NOTE]
> **Medium Risk**
> Billing projection changes affect entitlement logic for paid users; the environment rename must match GitHub configuration or production migrations will fail approval/secret lookup.
> 
> **Overview**
> **Clerk billing projection** no longer treats `past_due` subscription items as entitlements that can promote or extend paid tiers. Only **`active`** paid items drive upgrades; `past_due` paths keep the **current** paid tier (or return null for free users), including when reconciliation reports `subscriptionStatus: active` alongside a past-due upgrade item. Failed payment attempts no longer pull period end from Clerk items—they reuse the stored period end.
> 
> **CI/CD** renames the production GitHub Actions environment from `production` to **`Production – atlaris`** in the production DB migration workflow, deployment docs, and the architecture contract test so migrations use the same protected environment as other production jobs.
> 
> Regression tests cover reconciliation edge cases (active + past-due items, upcoming free after failed upgrade).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit cc591d9ee57002bc38e32c543055ffdb6cc0b3ba. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/features/billing/clerk-billing/projection.ts [hunk 1]
```diff
@@ -299,9 +296,12 @@ export function projectClerkBillingSource(
...
   if (
     source.paymentAttemptStatus === 'failed' ||
-    source.subscriptionStatus === 'past_due'
+    source.subscriptionStatus === 'past_due' ||
+    source.items.some(
+      (item) => isPaidTier(item.tier) && item.status === 'past_due',
+    )
   ) {
-    // Failed initial checkouts can include active paid items; never promote free users.
+    // Failed payments can include paid items; never use them to promote a tier.
```
[the remark this answers is not in the document]
Comment (saldanaj97) [reply src/features/billing/clerk-billing/projection.ts]: Fixed in cc591d9ee: active paid items are selected before past-due fallback, with a regression covering active Starter plus a separate past-due Pro upgrade.

[the remark this answers is not in the document]
Comment (saldanaj97) [reply src/features/billing/clerk-billing/projection.ts]: Fixed in cc591d9ee: past-due fallback now derives period end only from failed paid items matching the retained tier, then falls back to the current persisted period. Added the mixed upcoming-free regression.

File: src/features/billing/clerk-billing/projection.ts [hunk 2]
```diff
@@ -77,10 +77,7 @@ const PAID_TIER_RANK: Record<SubscriptionTier, number> = {
   pro: 2,
 };
 
-const ACTIVE_ITEM_STATUSES = new Set<ClerkSubscriptionItemStatus>([
-  'active',
-  'past_due',
-]);
+const ACTIVE_ITEM_STATUSES = new Set<ClerkSubscriptionItemStatus>(['active']);
```
[the remark this answers is not in the document]
Comment (saldanaj97) [reply src/features/billing/clerk-billing/projection.ts]: Fixed in cc591d9ee by removing the unreachable past-due ternaries and making active-item status handling explicit.

File: src/features/billing/clerk-billing/projection.ts [hunk 3]
```diff
@@ -77,10 +77,7 @@ const PAID_TIER_RANK: Record<SubscriptionTier, number> = {
...
 
 const TERMINAL_STATUSES = new Set<ClerkSubscriptionStatus>([
   'abandoned',
```
[the remark this answers is not in the document]
Comment (saldanaj97) [reply src/features/billing/clerk-billing/projection.ts:81]: No code change for this suggestion: Clerk defines past_due as a failed recurring charge, while the free plan has no recurring charge. The projection now requires the free item to be explicitly active and returns active directly.
